core/src/main/java/com/google/instrumentation/stats/MutableView.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2017, Google Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.instrumentation.stats;
+
+import com.google.instrumentation.common.Function;
+import com.google.instrumentation.common.Timestamp;
+import com.google.instrumentation.stats.View.DistributionView;
+import com.google.instrumentation.stats.ViewDescriptor.DistributionViewDescriptor;
+import com.google.instrumentation.stats.ViewDescriptor.IntervalViewDescriptor;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A mutable version of {@link View}, used for recording stats and start/end time.
+ */
+abstract class MutableView {
```

Comment:
I think we can simplify this class later, because it isn't part of the API.  For example, we probably don't need the factory methods.  We also don't need to pass in the distributions at the start, because the class is mutable.  I'm not suggesting any changes for this PR, though.